### DISCUSSION TITLE
Default linters

### DISCUSSION THREAD

**Discussion by cassepipe** [w0rp/ale#4387] [Q&A - Ask for help with problems] [answered]
The ale help says that below is the default dictionary that will be merged with user defined value of g:ale_linters. There's no mention of `c` or `cpp`.
```
  {                                                                                
  \   'apkbuild': ['apkbuild_lint', 'secfixes_check'],                             
  \   'csh': ['shell'],                                                            
  \   'elixir': ['credo', 'dialyxir', 'dogma'],                                    
  \   'go': ['gofmt', 'golint', 'gopls', 'govet'],                                 
  \   'hack': ['hack'],                                                            
  \   'help': [],                                                                  
  \   'inko': ['inko'],                                                            
  \   'json': ['jsonlint', 'spectral'],                                            
  \   'json': ['jsonlint', 'spectral', 'vscodejson'],                                                                                                 
  \   'json5': [],                                                              
  \   'jsonc': [],                                                              
  \   'perl': ['perlcritic'],                                                   
  \   'perl6': [],                                                              
  \   'python': ['flake8', 'mypy', 'pylint', 'pyright'],                        
  \   'rust': ['cargo', 'rls'],                                                 
  \   'spec': [],                                                               
  \   'text': [],                                                               
  \   'vader': ['vimls'],                                                       
  \   'vue': ['eslint', 'vls'],                                                 
  \   'zsh': ['shell'],                                                         
  \   'v': ['v'],                                                               
  \   'yaml': ['spectral', 'yaml-language-server', 'yamllint'],                 
  \} 
  ```
But ALE seems to be able to give me linting on C and C++ files without me setting any linter with that option. I first thought this was a because of the `compile_commands.json` file I generate with each build so I removed it, then I thought it was because of the `ale_parse_makefile` option I set, so I removed it, then I thought it might be due to the `ale_c_cc_options` but each time ALE was able to give  an error on say a unknown keyword.
I also tried `echo g:ale_linters` in vim and saw it was empty
How is that possible ?

**Comment by hsanson** [comment, marked as answer]
If you read further down in the help file you can find this statement:

>  All linters will be run for unspecified filetypes. 

If the resulting dictionary after merging the default one does not have an entry for c and c++ files then all available linters will be run for those files.

**Comment by w0rp** [comment]
@cassepipe Responding to an old question. If you want the behaviour of ALE only running exactly the linters you (or a plugin you installed) told it to run, you can set `g:ale_linters_explicit = 1`. The documented default configuration for different filetypes is only used to disable some linters by default, which are available and can be enabled, but are less likely to be helpful. For example, `gobuild` is a linter that runs `go test` on Go files, but it's probably not going to give good results, so we don't run that by default.

**Reply by cassepipe** [reply]
Good to know, thanks for the reply !
